### ISSUE TITLE
Unable to install miniupnpc-1.9

### ISSUE THREAD

**Issue by michalvalasek** [lbryio/lbry-setup#2]
I'm going through the install process on OS X 10.11.1 and I'm stuck on miniupnpc installation:
Download tarball: OK
Unpack tarball: OK
Run `sudo make`: OK
Run `sudo make install`:
```
cp miniupnpc.h miniupnpc.h.bak
sed 's/\(.*MINIUPNPC_API_VERSION\s\+\)[0-9]\+/\110/' < miniupnpc.h.bak > miniupnpc.h
install -d /usr/include/miniupnpc
install: mkdir /usr/include: Operation not permitted
make: *** [install] Error 71
```

I also tried to create the `/usr/include` folder manually, but I got the `Operation not permitted` msg again.

**Comment by jackrobison** [comment]
Have you turned off rootless mode? That's probably the problem, it prevents non Apple signed software from installing to anywhere but /usr/local, even as sudo. I'd first try installing it to /usr/local, and if that doesn't work try turning off rootless (link in the OS X readme), install, and then turn it back on.

**Comment by michalvalasek** [comment]
I'm sorry, I totally overlooked the instruction. I'll try that after work and let you know.

**Comment by jackrobison** [comment]
No problemo ;), and thanks for testing! Last night I added a .zip containing binaries, libs, and apps to the lbry-setup repo; that might be worth a go too. The binaries *should* work so long as they've got their corresponding lib folders in the same folder with them.

**Comment by kauffj** [comment]
If rootless mode continues to be required, can we do an early detect and
exit when it's not there?

On Wed, Dec 9, 2015 at 8:10 AM, jackrobison <notifications@github.com>
wrote:

> No problemo ;), and thanks for testing! Last night I added a .zip
> containing binaries, libs, and apps to the lbry-setup repo; that might be
> worth a go too. The binaries *should* work so long as they've got their
> corresponding lib folders in the same folder with them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lbryio/lbry-setup/issues/2#issuecomment-163225795>.
>



-- 

[image: TopScore] <http://www.usetopscore.com/>

Jeremy Kauffman, CEO
(267) 777-7592

[image: Twitter] <https://twitter.com/usetopscore> [image: Linkedin]
<https://www.linkedin.com/company/topscore-inc->

**Comment by jackrobison** [comment]
The binaries shouldn't need rootless mode, they don't need to install to /usr/include or /usr/bin. But yes, we can make it such that if a installer fails for lack of system write access it alerts you to the source of the problem rather than a more vague permission error.

**Comment by michalvalasek** [comment]
So I turned off the rootless mode and went on with the installation. This time I crashed at the 2nd step of "Building lbrycrd", i.e. `sudo ./configure`. Seems like I'm missing the EC part of OpenSSL:
```
...
checking openssl/ec.h usability... no
checking openssl/ec.h presence... no
checking for openssl/ec.h... no
configure: error: OpenSSL ec header missing
```
I tried to reinstall and re-link the OpenSSL via Homebrew but to no effect. I also exported variables `LDFLAGS` and `CPPFLAGS` as reccomended by openssl homebrew formula - still no effect.

I tried the compiled version too: the LBRY.app file doesn't seem to do anything (visible at least). I also launched the `lbrycrdd` binary - it's still running and doing something, but there's not output so I'm not sure about the status.
